File: attendance backend/controllers/Student.js
import { db } from "../config.js";

export const ShowStudents = async (req, res, next) => {
  var q =
    "select name,enrollment_no,department,section,semester from student where `department` = ? and `section` = ? and `semester` = ?;";

  var values = [
    req.body.department,
    req.body.section,
    req.body.semester
  ];

  db.query(q, [...values], (err, data) => {
    if (err) return res.json(err);
    return res.json(data);
  });
};

File: attendance/lib/models/student.dart
class Student {
  String? name;
  int? enrollmentNo;
  String? department;
  String? section;
  int? semester;

  Student(
      {this.name,
      this.enrollmentNo,
      this.department,
      this.section,
      this.semester});

  Student.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    enrollmentNo = json['enrollment_no'];
    department = json['department'];
    section = json['section'];
    semester = json['semester'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['name'] = this.name;
    data['enrollment_no'] = this.enrollmentNo;
    data['department'] = this.department;
    data['section'] = this.section;
    data['semester'] = this.semester;
    return data;
  }
}
